src/app/components/game/game.component.ts

```diff
@@ -264,24 +276,57 @@ export class GameComponent implements OnInit, OnChanges {
-                            7 - this.CURSOR_DATA.mouseDownOn.y
-                        )
-                    ) &&
-                    this.game.getPiece(
-                        new Square(
-                            this.CURSOR_DATA.mouseDownOn.x,
-                            7 - this.CURSOR_DATA.mouseDownOn.y
-                        )
-                    ).color === this.game.getTurn()
+                        this.CURSOR_DATA.mouseUpOn.y
                 ) {
-                    this.twoClickMove.attempting = true;
-                    this.twoClickMove.source = this.CURSOR_DATA.mouseUpOn;
+                    if (
+                        this.game.getPiece(
+                            new Square(
+                                this.CURSOR_DATA.mouseDownOn.x,
+                                7 - this.CURSOR_DATA.mouseDownOn.y
+                            )
+                        )?.color === this.game.getTurn()
+                    ) {
+                        this.twoClickMove.attempting = true;
+                        this.twoClickMove.source = this.CURSOR_DATA.mouseUpOn;
+                    }
                 } else {
-                    this.attemptMoveOnBoard();
+                    if (e.which === 1) {
+                        // mouse left click
+                        this.attemptMoveOnBoard();
+                    } else if (e.which === 3) {
+                        // mouse right click
+                        let unique = true;
+                        const newArrow = {
+                            fromSquare: new Square(
+                                this.CURSOR_DATA.mouseDownOn.x,
+                                this.CURSOR_DATA.mouseDownOn.y
+                            ),
+                            toSquare: new Square(
+                                this.CURSOR_DATA.mouseUpOn.x,
+                                this.CURSOR_DATA.mouseUpOn.y
+                            ),
+                            color: '#15781B',
+                            displacement: -5,
+                            pointerSize: 25,
+                            lineWidth: 15
+                        };
+                        const index = this.drawnArrows.findIndex(
+                            (arrow) =>
+                                arrow.fromSquare.toString() ===
+                                    newArrow.fromSquare.toString() &&
+                                arrow.toSquare.toString() ===
+                                    newArrow.toSquare.toString()
+                        );
+
+                        // delete if arrow already exists, otherwise add
+                        if (index !== -1) {
+                            this.drawnArrows.splice(index, 1);
+                            unique = false;
+                        }
+                        if (unique) {
+                            this.drawnArrows.push(newArrow);
+                        }
+                    }
```

Comment:
Bug report:

Go to move a piece by dragging it
Before releasing the piece to the destination square, press and hold right click
Notice the piece is in its dest and its src now, and keep dragging while holding both mouse buttons... still a bug.

This probably has to do with the locations in mouseDown/mouseUp where mouse click events happen. Maybe we need to trigger some of your code in different places within these functions, or we need to unset a value. I haven't dove into it. If you have trouble, let me know.

Reply:
Yeah code for the arrow should probably be in both mouseDown/mouseUp events. I considered doing that but it looked like I didn't need to. I guess I was wrong.

Reply:
Ok I just pushed a commit that fixed this issue and similar issues. I added two new properties `mouseLeftIsDown` and `mouseRightIsDown` to `CURSOR_DATA`. These two will probably also be useful in the future.